From e68996f7626e6b3544141d216ac662055ff73a64 Mon Sep 17 00:00:00 2001
From: Kazunori Kobayashi <kkobayas@igel.co.jp>
Date: Mon, 25 Jun 2012 17:30:48 +0900
Subject: [PATCH 07/39] base_videodec: set rowstride and chroma_byte_offset in
 caps

Images decoded by REL OMX in NV12 format must be aligned to a 32 line
boundary when stored in memory.
libshvio must be notified of the stride and the offset to the chroma plane
in order to properly display the video.
---
 omx/gstomx_base_videodec.c |    8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/omx/gstomx_base_videodec.c b/omx/gstomx_base_videodec.c
index 5914995..d5fdc38 100644
--- a/omx/gstomx_base_videodec.c
+++ b/omx/gstomx_base_videodec.c
@@ -43,6 +43,7 @@ settings_changed_cb (GOmxCore * core)
   guint width;
   guint height;
   guint32 format = 0;
+  gint32 stride, chroma_byte_offset;
 
   omx_base = core->object;
   self = GST_OMX_BASE_VIDEODEC (omx_base);
@@ -78,6 +79,9 @@ settings_changed_cb (GOmxCore * core)
       default:
         break;
     }
+
+    stride = param.format.video.nStride;
+    chroma_byte_offset = stride * param.format.video.nSliceHeight;
   }
 
   {
@@ -96,6 +100,10 @@ settings_changed_cb (GOmxCore * core)
       /* FIXME this is a workaround for xvimagesink */
       gst_structure_set (struc, "framerate", GST_TYPE_FRACTION, 0, 1, NULL);
 
+    gst_structure_set (struc, "rowstride", G_TYPE_INT, stride, NULL);
+    gst_structure_set (struc, "chroma_byte_offset", G_TYPE_INT,
+        chroma_byte_offset, NULL);
+
     gst_caps_append_structure (new_caps, struc);
 
     GST_INFO_OBJECT (omx_base, "caps are: %" GST_PTR_FORMAT, new_caps);
-- 
1.7.9.5

